test/e2e/specs/interactivity/directive-on-window.spec.ts

```diff
@@ -18,38 +18,47 @@ test.describe( 'data-wp-on-window', () => {
 		await utils.deleteAllPosts();
 	} );
 
-	test( 'callbacks should run whenever the specified event is dispatched', async ( {
-		page,
-	} ) => {
-		await page.setViewportSize( { width: 600, height: 600 } );
-		const counter = page.getByTestId( 'counter' );
-		await expect( counter ).toHaveText( '1' );
-	} );
-	test( 'the event listener is removed when the element is removed', async ( {
+	test( 'the event listener is attached when the element is added', async ( {
 		page,
 	} ) => {
 		const counter = page.getByTestId( 'counter' );
-		const isEventAttached = page.getByTestId( 'isEventAttached' );
 		const visibilityButton = page.getByTestId( 'visibility' );
 
+		// Initial value.
 		await expect( counter ).toHaveText( '0' );
-		await expect( isEventAttached ).toHaveText( 'yes' );
+
+		// Make sure the event listener is attached.
+		await page.waitForSelector(
+			'[data-testid="isEventAttached"]:has-text("yes")'
+		);
```

Comment:
The Playwright usually [discourages](https://playwright.dev/docs/api/class-page#page-wait-for-selector) the use of `page.waitForSelector`. Let's try the following:

```js
await page.getByTestId( 'isEventAttached' ).filter( { hasText: 'yes' } ).waitFor();
```

Reply:
Oh, thanks, George. 

Changed in https://github.com/WordPress/gutenberg/pull/58642/commits/7f3af2a6d6c659f08636fd611847f606683583ff.